### PR TITLE
Fix boolean condition in `publish.yaml`

### PR DESCRIPTION
**Pull request by mosuem**
As seen in https://github.com/dart-lang/native/actions/runs/6075163490/job/16511857492?pr=120, the `publish.yaml` seems to incorrectly retrieve the boolean value.

Fix it by using the JSON trick from `health.yaml`, tested using `publish_internal.yaml`.

CC @dcharkes 

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by devoncarew — APPROVED** [review]
lgtm

File: .github/workflows/publish_internal.yaml [hunk 1]
```diff
@@ -37,8 +40,8 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           ISSUE_NUMBER: ${{ github.event.number }}
           PR_LABELS: "${{ join(github.event.pull_request.labels.*.name) }}"
-        run: dart pkgs/firehose/bin/firehose.dart --validate
+        run: dart pkgs/firehose/bin/firehose.dart --validate ${{ fromJSON('{"true":"--use-flutter","false":"--no-use-flutter"}')[env.use-flutter] }}
```
Comment (HosseinYousefi) [inline .github/workflows/publish_internal.yaml:43]: Why is this necessary if env is always false?

Comment (HosseinYousefi) [reply .github/workflows/publish_internal.yaml:43]: Ah nevermind, the publish internal is for testing the trick.

Weird that the && doesn't work!

Comment (HosseinYousefi) [reply .github/workflows/publish_internal.yaml:43]: Question: didn't && and || and my PR work? The linked action seems to be using the old version (--use-flutter false) which is incorrect.
